tools/sigtest/src/main/java/com/sun/tdk/signaturetest/SignatureTest.java

```diff
@@ -1434,45 +1432,43 @@ private void checkAnnotations(MemberDescription base, MemberDescription test) {
 
         int bl = baseAnnotList.length;
         int tl = testAnnotList.length;
-        int bPos = 0;
-        int tPos = 0;
-
-        if (base != null && jdkExclude.isJdkClass(base.getDeclaringClassName())) {
-            return;
-        }
+        int requiredPos = 0;
+        int foundPos = 0;
 
-        if (test != null && jdkExclude.isJdkClass(test.getDeclaringClassName())) {
+        if (required != null && JDKExclude.isJdkClass(required.getDeclaringClassName())) {
+            // don't check excluded Annotation classes
             return;
         }
 
-        while ((bPos < bl) && (tPos < tl)) {
+        while ((requiredPos < bl) && (foundPos < tl)) {
             int comp = 0;
-            if (jdkExclude.isJdkClass(baseAnnotList[bPos].getName()) || 
-                    jdkExclude.isJdkClass(testAnnotList[tPos].getName())) {
-                comp = baseAnnotList[bPos].getName().compareTo(testAnnotList[tPos].getName());
-            } else {
-                comp = baseAnnotList[bPos].compareTo(testAnnotList[tPos]);
+            if (JDKExclude.isJdkClass(baseAnnotList[requiredPos].getName()) ||
+                    JDKExclude.isJdkClass(testAnnotList[foundPos].getName())) {
+                // don't check excluded Annotation classes
+                return;
```

Comment:
Should this be `continue` instead of `return`?  Otherwise, you won't check the rest of the annotations.

Reply:
Thanks again @KyleAure as this change led to a more correct `checkAnnotations` implementation as we made the suggested fix and also changed to validate JDK annotations by checking that the implementation uses the same JDK annotation class name.